GiniVision/Classes/Core/CameraViewController.swift

```diff
@@ -574,6 +518,81 @@ extension CameraViewController {
         }
     }
     
+    fileprivate func setupCamera(giniConfiguration: GiniConfiguration) {
+        self.camera = Camera(giniConfiguration: giniConfiguration) {[weak self] error in
+            if let error = error {
+                switch error {
+                case .notAuthorizedToUseDevice:
+                    addNotAuthorizedView()
+                default:
+                    if GiniConfiguration.DEBUG { cameraState = .valid; addDefaultImage() }
+                }
+                self?.failureBlock?(error)
+            }
+        }
+        self.camera?.didDetectQR = {[weak self] qrDocument in
+            if self?.detectedQRCodeDocument != qrDocument {
+                self?.detectedQRCodeDocument = qrDocument
+                self?.didPick(qrDocument)
+            }
+        }
+    }
+    
+    @objc fileprivate func captureImage(_ sender: AnyObject) {
+        guard let camera = camera else {
+            Log(message: "No camera initialized", event: .warning)
+            return
+        }
+        
+        if GiniConfiguration.DEBUG {
+            // Retrieve image from default image view to make sure image
```

Comment:
Retrieves **the** image from **the** image view to make sure **the** image!!!!!

Articles!!!